File: chae-dahee/ch06/18310.js
let input = require("fs").readFileSync('/dev/stdin').toString().trim().split("\n");

const n = parseInt(input[0]);
const house = input[1].split(" ").map(Number)

function Antenna(house){
    // house.sort((a, b) => b-a);
    // return  position = house[Math.floor(house.length / 2)];

    house.sort((a, b) => a-b);
    const  position = Math.floor(house.length / 2);
    return house.length % 2 === 0? house[position-1] : house[position];
}

console.log(Antenna(house));

File: woogieon8on/chap6/boj/18310.py
N = int(input())
house = list(map(int, input().split()))

# 모든 집까지의 거리의 총합
# : 안테나를 모든 집들 중 중간에 위치한 집에 설치한 경우 최소
house.sort()
antenna = house[int((N - 1) / 2)] # 안테나를 설치할 수 있는 위치 값으로 여러 개의 값이 도출될 경우 가장 작은 값 출력

print(antenna)